### ISSUE TITLE
Model for automated data annotation in context of traffic participants

### ISSUE THREAD

**Issue by IlkayW** [CVHub520/X-AnyLabeling#795]
### Search before asking

- [X] I have searched the X-AnyLabeling [Docs](https://github.com/CVHub520/X-AnyLabeling/tree/main/docs) and [issues](https://github.com/CVHub520/X-AnyLabeling/issues) and found no similar questions.

### Question

Hello,

I'm currently working on my PhD thesis. One part of this thesis is a model-based auto annotation algorithm, which I am evaluating on the CityScapes dataset. The desired object classes are: person, bicycle, car, motorcycle, bus, train and truck.

In order to have a good comparison with other approaches, I want to know which model(s) do you recommend for best auto annotation with X-AnyLabeling?

Greetings,
Ilkay

**Comment by CVHub520** [comment]
Dear Ilkay,

Thanks for reaching out! I'm glad to hear you're considering X-AnyLabeling for your PhD thesis research on model-based auto-annotation.

The Cityscapes dataset is a popular choice for semantic segmentation. Here are some resources that you might find helpful:

* **Papers with Code - Cityscapes Dataset:** This website provides a comprehensive overview of the Cityscapes dataset, including  papers, code, and metrics commonly used for semantic segmentation on Cityscapes. It's a great place to start to learn about the state-of-the-art and find inspiration for your research: [https://paperswithcode.com/dataset/cityscapes](https://paperswithcode.com/dataset/cityscapes)

While I can't directly recommend specific models within X-AnyLabeling, exploring these resources should help you identify suitable architectures for your use case. X-AnyLabeling is a versatile annotation tool that can be used with a variety of deep-learning models for serveral tasks.

Feel free to ask if you have any other questions about using X-AnyLabeling for your research.

Best regards,
The X-AnyLabeling Maintainer

**Comment by IlkayW** [comment]
If you are interested in the results:


Method:
The images from the train and validation set were automatically annotated with my auto annotation tool, with X-AnyLabeling using YOLOv5/YOLOv8 SAHI and YOLOv11x and additionally I used the Roboflow auto annotation tool.
For the comparison, I've converted the polygons from semantic segmentation to bounding boxes. In that way, I get the CityScapes detection dataset, which are considered as ground truth.
Finally, I've compared the auto annotation with the ground truth annotations.
Object classes: person, bicycle, car, motorcycle, bus, train, truck

Results:
My tool first version: IW_AA_v1
Yours: X-SAHI_YOLOv5s/X-SAHI_YOLOv8s/X-YOLOv11x
Roboflow: RF_COCOv3

TP=True Positive, FC=False Classified, FL=False Located, FP=False Positive, FN=False Negative
IoU for TP or FC >0.7
IoU for FL <0.7 and >0.4

![Image](https://github.com/user-attachments/assets/fc8f2163-4b37-4ddf-a879-6ef1b8eae394)
